### PR TITLE
Event metrics

### PR DESCRIPTION
**Pull request by mlw**
Add metrics for event counts and event processing times.

Current design passes a block to ES client message handlers to be called when they're done processing and event.

Alternative considered was to have the ES client message handlers return a disposition (instead of calling a block) and leave the work to the base class. This could work but would add some complexity for clients that processed messages async. Given there are only a few clients, it wouldn't be a huge burden to go this route in the future if it provided advantages.

### CODE REVIEW

File: Source/santad/SantadDeps.mm [hunk 1]
```diff
@@ -109,21 +112,29 @@
     exit(EXIT_FAILURE);
   }
 
-  return std::make_unique<SantadDeps>([configurator metricExportInterval], esapi, std::move(logger),
-                                      control_connection, compiler_controller, notifier_queue,
-                                      syncd_queue, exec_controller, prefix_tree);
+  // TODO: `SNTMetricSet sharedInstance` should be injected
```
Comment (russellhancox) [inline Source/santad/SantadDeps.mm:115]: This is done? Or am I missing something?

File: Source/santad/Metrics.mm [hunk 2]
```diff
@@ -13,26 +13,105 @@
...
+      return nil;
+  }
+}
+
+const NSString *EventDispositionToString(EventDisposition d) {
+  switch (d) {
+    case EventDisposition::kDropped: return kEventDispositionDropped;
+    case EventDisposition::kProcessed: return kEventDispositionProcessed;
+    default:
+      [NSException raise:@"Invalid processor" format:@"Unknown processor value: %d", d];
```
Comment (russellhancox) [inline Source/santad/Metrics.mm:90]: s/processor/disposition

**Review by russellhancox — CHANGES_REQUESTED** [review]

File: Source/santad/SantadDeps.mm (hunk 1 above)
Comment (mlw) [reply Source/santad/SantadDeps.mm]: Oops, old comment. Removed.

File: Source/santad/Metrics.mm (hunk 2 above)
Comment (mlw) [reply Source/santad/Metrics.mm]: Fixed

**Review by russellhancox — APPROVED** [review]

**Pull request merged**